File: Data Structures & Algorithms/two-integer-sum/submission-6.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // edge cases

        var map = new Dictionary<int, int>();

        for(int i =0; i<nums.Length; i++) {
            var find = target - nums[i];
            if(map.ContainsKey(find) && map[find] != i) {
                return new int[] {map[find], i};
            } else {
                map.Add(nums[i], i);
            }
        }

        return new int[] {-1, -1};
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-6.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        if not nums:
            return False
        
        hs = set()
        
        for num in nums:
            if num in hs:
                return True
            hs.add(num)
        
        return False

        